pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -687,6 +687,13 @@ func ApplyReplacements(spec *v1.TaskSpec, stringReplacements map[string]string,
 		container.ApplyStepTemplateReplacements(spec.StepTemplate, stringReplacements, arrayReplacements)
 	}
 
+	// Apply variable expansion to podTemplate fields.
+	if spec.PodTemplate != nil {
+		for key, value := range spec.PodTemplate.NodeSelector {
```

Comment:
I don't know the code very well there, but that only applies to the NodeSelector in the pod template; would there be more podtemplate specs that would needs to be applied here?

Reply:
@chmouel Thanks for taking a look - yes I think it would make sense to support variable expansion in all of the podTemplate fields. Currently working on that and will push it up (hopefully) soon with some other updates.